src/graphql/models/User.js

```diff
@@ -158,6 +159,63 @@ const User = new GraphQLObjectType({
         'If not null, the user is blocked with the announcement in this string.',
       type: GraphQLString,
     },
+
+    badges: {
+      type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(Badge))),
+      description: 'Badges awarded to the user.',
+      resolve: (user) => user.badges || [],
+    },
+
+    majorBadgeBorderUrl: {
+      type: GraphQLString,
+      description: 'returns badge background image url',
+      resolve: async (user, atgs, { loaders }) => {
+        const displayItem = user.badges.find({ isDisplay: true });
+        if (displayItem == null) {
+          return null;
+        }
+        const badgeId = displayItem.id;
+        const badgeInfo = loaders.docLoader.load({
+          index: 'badges',
+          id: badgeId,
+        });
+        return badgeInfo.borderImage;
+      },
+    },
+
+    majorBadgeImageUrl: {
+      type: GraphQLString,
+      description: 'returns badge background image url',
+      resolve: async (user, atgs, { loaders }) => {
+        const displayItem = user.badges.find({ isDisplay: true });
+        if (displayItem == null) {
+          return null;
+        }
+        const badgeId = displayItem.id;
+        const badgeInfo = loaders.docLoader.load({
+          index: 'badges',
+          id: badgeId,
+        });
+        return badgeInfo.icon;
+      },
+    },
+
+    majorBadgeName: {
+      type: GraphQLString,
+      description: 'returns badge background image url',
+      resolve: async (user, atgs, { loaders }) => {
```

Comment:
nit: typo. Also for other resolvers
```suggestion
      resolve: async (user, args, { loaders }) => {
```